### ISSUE TITLE
Please improve output from Az Policy commands

### ISSUE THREAD

**Issue by kilasuit** [Azure/azure-powershell#10593]
## Description
The items returned from Azure Policy commands are horrific to work with because of how the underlying service has been implemented with a name of a policy not matching the actual name you'd see in the portal.

As per the below you can see that there is a `Name` which is really an ID and the `displayName` is a sub property of Properties that really should be properly exposed at the top level of the object returned in this command and the other Policy commands
![image](https://user-images.githubusercontent.com/6355225/69431175-1e3a9080-0d2f-11ea-91be-97a29fb5814a.png)

Also why the heck are the commands split across 2 different modules - that makes no sense
## Environment data

``` powershell
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.0.0-preview.5
PSEdition                      Core
GitCommitId                    7.0.0-preview.5
OS                             Microsoft Windows 10.0.19025
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

## Module versions

```powershell
 gcm Get-AzPolicy*

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Cmdlet          Get-AzPolicyAlias                                  1.6.0      Az.Resources
Cmdlet          Get-AzPolicyAssignment                             1.6.0      Az.Resources
Cmdlet          Get-AzPolicyDefinition                             1.6.0      Az.Resources
Cmdlet          Get-AzPolicyEvent                                  1.1.2      Az.PolicyInsights
Cmdlet          Get-AzPolicyRemediation                            1.1.2      Az.PolicyInsights
Cmdlet          Get-AzPolicySetDefinition                          1.6.0      Az.Resources
Cmdlet          Get-AzPolicyState                                  1.1.2      Az.PolicyInsights
Cmdlet          Get-AzPolicyStateSummary                           1.1.2      Az.PolicyInsights
```

**Comment by mentat9** [comment]
Sorry that the naming is confusing. For policy objects, the "Name" property displayed by the portal is actually the "displayName" property as implemented by the service. The fields provided by the Powershell cmdlets and all other SDKs match what is actually implemented by the service. The portal chose to present displayName instead of name. We definitely understand this is confusing at first, but once you understand how the portal works, it's not too hard to deal with. The portal displays and filters on something called "Name" which is actually "displayName". Since it uses "Name" for "displayName" it doesn't expose the real "name" field, instead filling it with a guid when creating a new policy object. The name field must be unique, and the object's Id is its name qualified by namespace and resource type. DisplayName can be anything and is not required to be unique.

From Powershell and the other SDKs, you can separately control name and displayName. Name is not required to be a guid: it can be anything you want that isn't already taken. If you want the portal to match Powershell, you can always just make name and displayName the same.

So to summarize:
1. The top-level name can't be what the portal shows, because it must be unique.
2. The portal doesn't give you access to the real name.
3. The field names exposed by Powershell and other SDKs are the real ones.

Regarding the reasons for policy cmdlets to not all be in the same module: it's for the same reason that all Azure cmdlets are not in the same module. When cmdlets are in different modules, their code, behavior, and API versions can be updated separately. It may not appear logical, since we think of Policy as one thing, but under the covers it's many things.

**Comment by kilasuit** [comment]
@mentat9 - what I am asking for is that the displayname come up a level in the properties returned to the user from the API call that is completed.

That's easily enough done in powershell anyway as an additional step that I really should not need to do on every single use of these API's

It's not a great UserExperience and that's what I am asking to change. If that needs to change in the SDK then I will look to raise this there instead

**Comment by mentat9** [comment]
@kilasuit, thanks for clarifying. Unfortunately, what you are asking for is a significant breaking change. The object layout is basic to the operation of policy and is consistent with many/most of the other ARM cmdlets, so changing it for client-side convenience is a big deal.

I've investigated this further and have some ideas how we might improve in this area; however, even if it works out it will take a long time. The breaking change process for Powershell requires multiple releases that need to be far enough apart to give customers time to respond to deprecations before they happen.

We've heard your feedback loud and clear and are working out how to make improvements in this area, but unfortunately I can't give you any kind of promise or ETA: there are simply too many interdependent steps and too much uncertainty at this point.

Thank-you again for your feedback.
